base/abstractarraymath.jl

```diff
@@ -81,6 +81,29 @@ end
 
 squeeze(A::AbstractArray, dim::Integer) = squeeze(A, (Int(dim),))
 
+"""
+    squeeze(A)
+
+Remove all singleton dimensions from array `A`.
+
+```jldoctest
+julia> a = reshape(collect(1:4),(2,2,1,1))
+2×2×1×1 Array{Int64,4}:
+[:, :, 1, 1] =
+ 1  3
+ 2  4
+
+julia> squeeze(a)
+2×2 Array{Int64,2}:
+ 1  3
+ 2  4
+```
+"""
```

Comment:
I'm not sure what I would write?  This function is incredibly straightforward; it removes all singleton dimensions.

Looking back over the previous discussion, it looks like the concern is that you may pass in an `NxMxL` matrix expecting `L` to equal 1, but then being surprised when `M` also happened to equal 1 and thus the result is a 1-dimensional array.  I'm not sure how to make the docstring more explicit that such a thing will happen; that is, after all, the entire point of this function.